screens/DoubleSearch.js

```diff
@@ -0,0 +1,220 @@
+import React, { useEffect } from "react";
+import PropTypes from "prop-types";
+import { View, Text, StyleSheet, AsyncStorage} from "react-native";
+import { Icon, Button } from "native-base";
+import SearchableDropdown from "react-native-searchable-dropdown";
+import { MapData } from "../components/MapData";
+import { sgwRooms } from "../constants/sgwRooms";
+import { buildingData } from "../constants/buildingData";
+import { DoubleSearchSVG } from "../assets/DoubleSearchSVG.js";
+
+
+
+function fetchData() {
+    const searchInfo = MapData({ passBuildingName: "", buildingName: true, classRooms: true, departments: true, services: true, accesibility: false, flatten: true }, sgwRooms(), buildingData());
+    return searchInfo;
+}
+
+/**
+ * US12 - As a user, I want to be able to select a destination building by clicking on it.
+ * US14 - As a user, I should be able to set my current location as the starting point.
+ *
+ * The following function renders a preference menu with 2 search bars. The "from" conatains the current location which 
+ * is set automatically (but can be modified) and the "to" contains the destination
+ */
+DoubleSearch.propTypes ={
+    navigation: PropTypes.object
+};
+
+function DoubleSearch(props) {
+
+    const [data, setData] = React.useState();
+    const [to, setTo] = React.useState("");
+    const [from, setFrom] = React.useState("");
+
+    const goToBackToMoreDetails = () => {
+        props.navigation.goBack();
+    };
+    const getDirectionScreen = () => {
+        props.navigation.navigate("MapDirections");
+    };
+
+    const selectDestinationPath = () => {
+        return props.navigation.getParam("destinationName", "null");
+    };
+
+    //TODO: The new "From" location entered by the user will be stored in the asyncStorage with this key
+    AsyncStorage.setItem("fromLocation",from.toString());
+    // The new "To" destination can also be retrieved from the asyncStorage with this key
+    AsyncStorage.setItem("destination",to.toString());
+
+    useEffect(() => {
+        const intervalId = setInterval(() => {
+            setData(fetchData());
+        }, 1);
+        return () => clearInterval(intervalId);
+    });
+
+    return (
```

Comment:
This return will need refactoring after all the functionality are working. But, here the fact that we have a search bar in the Home Map and we rewriting the code is not so good. It would make the code more readable and perhaps more efficient if we were reusing the Search Component.  No need to make any changes for this PR!

Reply:
Noted!